dll/src/ReflectiveLoader.c

```diff
@@ -95,6 +97,30 @@ RDIDLLEXPORT ULONG_PTR WINAPI ReflectiveLoader( VOID )
 	ULONG_PTR uiValueE;
 	DWORD dwProtect;
 
+	PVOID pNtdllBase = NULL, pKernel32 = NULL;
+
+	// The NTDLL exported functions our loader needs. This array of Syscall structures will be completed later.
+#ifdef ENABLE_STOPPAGING
+	Syscall Syscalls[4];
+#else
+	Syscall Syscalls[3];
+#endif
+	Syscalls[0].dwCryptedHash = NTALLOCATEVIRTUALMEMORY_HASH;
+	Syscalls[0].dwNumberOfArgs = 6;
+	Syscalls[0].hooked = FALSE;
+	Syscalls[1].dwCryptedHash = NTPROTECTVIRTUALMEMORY_HASH;
+	Syscalls[1].dwNumberOfArgs = 5;
+	Syscalls[1].hooked = FALSE;
+	Syscalls[2].dwCryptedHash = NTFLUSHINSTRUCTIONCACHE_HASH;
+	Syscalls[2].dwNumberOfArgs = 3;
+	Syscalls[2].hooked = FALSE;
+#ifdef ENABLE_STOPPAGING
+	Syscalls[3].dwCryptedHash = NTLOCKVIRTUALMEMORY_HASH;
+	Syscalls[3].dwNumberOfArgs = 4;
+	Syscalls[3].hooked = FALSE;
+#endif

```

Comment:
These indexes are a little difficult to track while reading through the source code. The indexes also can't be constants because of how `ColdGate.c` uses it's own array of syscalls to define the ones it needs.

It would help to define them individually:

```suggestion
	// The NTDLL exported functions our loader needs. This array of Syscall structures will be completed later.
    Syscall NtAllocateVirtualMemorySyscall = { NTALLOCATEVIRTUALMEMORY_HASH, 6, FALSE };
    Syscall NtProtectVirtualMemorySyscall = { NTPROTECTVIRTUALMEMORY_HASH, 5, FALSE };
    Syscall NtFlushInstructionCacheSyscall = { NTFLUSHINSTRUCTIONCACHE_HASH, 3, FALSE };
#ifdef ENABLE_STOPPAGING
    Syscall NtLockVirtualMemorySyscall = { NTLOCKVIRTUALMEMORY_HASH, 4, FALSE };
#endif
```

Note that technically per the [C99](https://www.dii.uchile.cl/~daespino/files/Iso_C_1999_definition.pdf) spec, you could also omit the `FALSE` at the end if you want to favor brevity over verbosity.

This same suggestion goes for what is written in ColdGate as well.

Reply:
Thanks for the suggestions, I addressed this in 10fe5f7ad6f28f68f257533e36f94018c4d68ab1